applications/FluidDynamicsApplication/python_scripts/trilinos_navier_stokes_embedded_solver.py

```diff
@@ -118,89 +106,94 @@ def ImportModelPart(self):
         self.distributed_model_part_importer.ImportModelPart()
         ## Sets DENSITY, VISCOSITY and SOUND_VELOCITY
 
-        KratosMultiphysics.Logger.PrintInfo("NavierStokesMPIEmbeddedMonolithicSolver","MPI model reading finished.")
+        KratosMultiphysics.Logger.PrintInfo(self.__class__.__name__,"MPI model reading finished.")
 
     def PrepareModelPart(self):
         super(NavierStokesMPIEmbeddedMonolithicSolver,self).PrepareModelPart()
         ## Construct MPI the communicators
         self.distributed_model_part_importer.CreateCommunicators()
 
-    def AddDofs(self):
-        super(NavierStokesMPIEmbeddedMonolithicSolver, self).AddDofs()
-
-        KratosMultiphysics.Logger.PrintInfo("NavierStokesMPIEmbeddedMonolithicSolver","DOFs for the VMS Trilinos fluid solver added correctly.")
-
-    def Initialize(self):
-        ## Construct the communicator
-        self.EpetraCommunicator = KratosTrilinos.CreateCommunicator()
-
-        ## Get the computing model part
-        self.computing_model_part = self.GetComputingModelPart()
-
-        ## If needed, create the estimate time step utility
-        if (self.settings["time_stepping"]["automatic_time_step"].GetBool()):
-            self.EstimateDeltaTimeUtility = self._GetAutomaticTimeSteppingUtility()
-
-        # Set the time discretization utility to compute the BDF coefficients
-        time_order = self.settings["time_order"].GetInt()
-        if time_order == 2:
-            self.time_discretization = KratosMultiphysics.TimeDiscretization.BDF(time_order)
+    def _GetEpetraCommunicator(self):
+        if not hasattr(self, '_epetra_communicator'):
+            self._epetra_communicator = KratosTrilinos.CreateCommunicator()
+        return self._epetra_communicator
+
+    def _CreateScheme(self):
+        domain_size = self.GetComputingModelPart().ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
+        # Cases in which the element manages the time integration
+        if self.element_integrates_in_time:
+            # "Fake" scheme for those cases in where the element manages the time integration
```

Comment:
if you need a "fake" scheme why not using the baseclass (`Scheme`) here?

Reply:
Because the base class `Scheme` does not implement the `Update`. Maybe I should put a more explicative comment as in the serial solver.

Reply:
The comment was already there (line below)